shortcode.php

```diff
@@ -330,3 +330,110 @@ function geoip_detect_shortcode_user_info($attr) {
     return geoip_detect2_shortcode_user_info_wpcf7('', 'geoip_detect2_user_info', true);
 }
 add_shortcode('geoip_detect2_user_info', 'geoip_detect_shortcode_user_info');
+
+/**
+ *
+ * Content Hiding
+ * 
+ * Uses an enclosing shortcode (i.e. [shortcode]Content[/shortcode]). Shortcode attributes can be as follows:
+ * "timezone", "continent", "country", "region"/"state", "city", and "and-not".
+ * 
+ * Each attribute may only appear once in a shortcode, and each attribute can only contain a single value, or
+ * the shortcode will break, showing the contents no matter what.
+ *
+ * Examples:
+ *
+ * `[geoip_detect2_hide_unless country="US" state="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="TX"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * 	    - OR -
+ * `[geoip_detect2_hide_unless country="US" region="Texas"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state.
+ * 
+ * `[geoip_detect2_hide_unless country="US" state="TX" and-not="Houston"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state,
+ * and the specified exclusion is not found in any possible GeoIP Detect field (e.g. visitor's country,
+ * city, timezone, etc.).
+ * 
+ * `[geoip_detect2_hide_unless contenant="North America"]<h1>Title</h1>[/geoip_detect2_hide_unless]`
+ * Only displays the inner content if the specified attributes match the visitor's contenant.
+ * 
+ */
+function geoip_detect2_shortcode_hide_unless($atts, $content = null) {
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and-not' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
+    $criteria_test_flag = true;       // If set to false, nothing will display
+    
+    /* Attribute Conditions */
+    // Timezone
+    if ($atts_array['timezone'] != null) {
+        if ($info->location->timeZone && $atts_array['timezone'] != $info->location->timeZone) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Continent
+    if ($atts_array['continent'] != null) {
+        if ($info->continent->code && $atts_array['continent'] != $info->continent->code) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Country
+    if ($atts_array['country'] != null) {
+        if ($info->country->name && $atts_array['country'] != $info->country->name
+            && $info->country->isoCode && $atts_array['country'] != $info->country->isoCode) {
+            $criteria_test_flag = false;
+        }
+    }
+
+    // Region/State
```

Comment:
I would prefer to use `most-specific-subdivision` as parameter name as it explains what is going on the Maxmind side of things. It is state for US/Canada, region for otherwise?

Reply:
Can do. I either copied `mostSpecificSubdivision` from above in the same file, or I found it through the settings page of a Wordpress install.

I intended "state" and "region" attributes to both reference the same thing, depending on the location/preference of the user.

Reply:
Sorry that's not what I meant, either. `mostSpecificSubdivision` is the correct syntax for the PHP side of things. I was talking about the shortcode parameter name. 

But you're right, using _ instead of - everywhere is more consistent with how I'm currently treating shortcodes.

Reply:
Solved by my commit.